### PR TITLE
Feature/gt 174 unity analytics implementation

### PR DESCRIPTION
**Pull request by egecnr**
Can you check the intellisense issues? Is it just me or does everyone have it?

### CODE REVIEW

File: Assets/Scripts/HUD/EndOfDayReport.cs [hunk 1]
```diff
@@ -51,13 +56,53 @@ void Start()
...
         Cursor.lockState = CursorLockMode.None;
         Cursor.visible = true;
+        Time.timeScale = 0f;
         
     }
 
+    private void writePlayFabData()
+    {
+        playFabManager.SendLeaderBoard(playerReportData.GetTheSuccessfulMinigameNumber());
+        playFabManager.WriteCustomPlayerEvent("Distance_travelled_per_game",new Dictionary<string, object> {
```
Comment (dbecu) [inline Assets/Scripts/HUD/EndOfDayReport.cs:70]: Would it be possible if you make a method out of these? And then the different values as that method's params?

File: Assets/Scripts/MiniGameManager.cs [hunk 2]
```diff
@@ -47,6 +51,12 @@ public void StartGame(GameObject miniGamePrefab)
 
     public IEnumerator StopGame(GameObject go)
     {
+        miniGameTime -= TimerCountdown.Instance.GetRemainingTime();
+        //personalize this later to make it for each mini game.
+        playFabManager.WriteCustomPlayerEvent($"{playFabManager.returnPrefabTaskName(this.InteractableObject.GamePrefab.name.ToString())}_FinishedInSeconds", new Dictionary<string, object>
```
Comment (dbecu) [inline Assets/Scripts/MiniGameManager.cs:58]: Could use method advised previously here

**Review by dbecu — COMMENTED** [review]
Overall looks great! It works super good, good job! I made a comment about reducing redundancy, but up to you if you wanna implement it

File: Assets/Scripts/HUD/EndOfDayReport.cs (hunk 1 above)
Comment (egecnr) [reply Assets/Scripts/HUD/EndOfDayReport.cs:70]: That is in the end WriteCustomPlayerEvent. I created that method to make  events it is not built in

**Pull request merged**